From 297d05f56289dac3bd1466492cfa930c3b6bd81a Mon Sep 17 00:00:00 2001
From: notsyncing <song.fc@gmail.com>
Date: Sun, 12 May 2019 13:58:26 +0800
Subject: [PATCH 2/3] org.ifaa.android.manager: Add IIFAAService for Alipay
 fingerprint payment

  * The org.ifaa.android.manager.IIFAAService has been moved
    from the SoterService.apk to MIUI framework, at least
    since MIUI 9.4.26. Adding this interface will fix the
    SoterService crashing when launching Alipay.

Change-Id: Ic97467eb0a8fe92b49e0edbedd56f1866c5fa01e
---
 org.ifaa.android.manager/Android.bp                  |  4 ++++
 .../src/org/ifaa/android/manager/IIFAAService.aidl   | 12 ++++++++++++
 2 files changed, 16 insertions(+)
 create mode 100644 org.ifaa.android.manager/src/org/ifaa/android/manager/IIFAAService.aidl

diff --git a/org.ifaa.android.manager/Android.bp b/org.ifaa.android.manager/Android.bp
index 3e637f4..a77c477 100644
--- a/org.ifaa.android.manager/Android.bp
+++ b/org.ifaa.android.manager/Android.bp
@@ -19,5 +19,9 @@ java_library {
     installable: true,
     srcs: [
         "src/**/*.java",
+        "src/**/I*.aidl",
     ],
+    aidl: {
+        local_include_dirs: ["src"],
+    },
 }
diff --git a/org.ifaa.android.manager/src/org/ifaa/android/manager/IIFAAService.aidl b/org.ifaa.android.manager/src/org/ifaa/android/manager/IIFAAService.aidl
new file mode 100644
index 0000000..80937fe
--- /dev/null
+++ b/org.ifaa.android.manager/src/org/ifaa/android/manager/IIFAAService.aidl
@@ -0,0 +1,12 @@
+package org.ifaa.android.manager;
+
+interface IIFAAService {
+    byte[] processCmd_v2(in byte[] param);
+    int[] getIDList(int bioType);
+    int faceEnroll(String sessionId, int flags);
+    int faceUpgrade(int action, String path, int offset, in byte[] data, int data_len);
+    int faceAuthenticate_v2(String sessionId, int flags);
+    int faceCancel_v2(String sessionId);
+    byte[] faceInvokeCommand(in byte[] param);
+    int faceGetCellinfo();
+}
-- 
2.17.1

